bindings/kvp/src/main/java/org/n52/sos/decode/kvp/AbstractKvpDecoder.java

```diff
@@ -355,23 +357,43 @@ private TemporalFilter createTemporalFilterFromValue(String value, String valueR
             temporalFilter.setOperator(TimeOperator.TM_Equals);
             temporalFilter.setTime(ti);
         } else if (times.length == 2) {
-            DateTime start = DateTimeHelper.parseIsoString2DateTime(times[0]);
-            // check if end time is a full ISO 8106 string
-            int timeLength = DateTimeHelper.getTimeLengthBeforeTimeZone(times[1]);
-            DateTime origEnd = DateTimeHelper.parseIsoString2DateTime(times[1]);
-            DateTime end = DateTimeHelper.setDateTime2EndOfMostPreciseUnit4RequestedEndPosition(origEnd, timeLength);
+            LOGGER.debug("Parsing temporal filter, start: {}, end: {}", times[0], times[1]);
+            DateTime start = null;
+            DateTime end = null;
+            if (times[0].startsWith(ISO_8601_DURATION_INDICATOR)) {
```

Comment:
It handles the inclusiveness of end times differently than 52N currently does. An end time of for example 2015-07 will be interpreted as 2015-07-01T00:00:00, instead of 2015-07-31T23:59:59.